queries/systemtap/highlights.scm

```diff
@@ -0,0 +1,156 @@
+(identifier) @variable
+
+(preprocessor_macro_definition
+  name: (identifier) @define)
+
+(preprocessor_macro_expansion) @function.macro
+
+(preprocessor_constant) @constant.macro
+
+(number) @number
+(string) @string
+(escape_sequence) @string.escape
+
+[
+  (script_argument_string)
+  (script_argument_number)
+] @constant
+
+(probe_point_component) @function
+
+(function_definition
+  name: (identifier) @function)
+
+(parameter
+  name: (identifier) @parameter)
+
+(type) @type.builtin
+
+(aggregation_operator) @function.macro
+
+(member_expression
+  member: (identifier) @field)
+
+(call_expression
+  function: (identifier) @function.call)
+
+(
+  (call_expression
+    function: (identifier) @function.builtin)
+  (#any-of? @function.builtin
+    "print" "printd" "printdln" "printf" "println"
+    "sprint" "sprintd" "sprintdln" "sprintf" "sprintln")
+)
+
+(
+  (identifier) @variable.builtin
+  (#lua-match? @variable.builtin "^\$+[0-9A-Z_a-z]+\$*$")
+)
+
+(shebang_line) @preproc
+
+(comment) @comment @spell
+
+[
+  "!"
+  "!="
+  "!~"
+  "$"
+  "$$"
+  "%"
+  "%="
+  "&"
+  "&&"
+  "&="
+  "*"
+  "*="
+  "+"
+  "++"
+  "+="
+  "-"
+  "--"
+  "-="
+  "->"
+  "."
+  ".="
+  "/"
+  "/="
+  ":"
+  "<"
+  "<<"
+  "<<<"
+  "<<="
+  "<="
+  "="
+  "=="
+  "=~"
+  ">"
+  ">="
+  ">>"
+  ">>="
+  "?"
+  "^"
+  "^="
+  "|"
+  "|="
+  "||"
+  "~"
+] @operator
+
+[
+  ","
+  (null_statement)
+] @punctuation.delimiter
+
+[
+  "("
+  ")"
+  "["
+  "]"
+  "{"
+  "}"
+] @punctuation.bracket
+
+[
+  "break"
+  "continue"
+  "delete"
+  "global"
+  "limit"
+  "next"
+  "private"
```

Comment:
Could be `@type.qualifier`

Reply:
I’m wary of using `@type` here since `private` can be used only for variable declarations and function definitions. Moreover, SystemTap has no user-defined types.

Reply:
private is a qualifier though, isn't it? `@type.qualifier` really works for any sort of qualifier

Reply:
Is it the convention in nvim-treesitter to be able to use `@type.qualifier` to qualify entities other than types? If so, then I will update the capture.

Reply:
One day it will be renamed to something more sensible…